src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -218,47 +212,30 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
                 if (!hasErrors)
                     CheckFeatureAvailability(innerExpression, MessageID.IDS_FeatureTypePattern, diagnostics);
 
-                if (hasSuppression(expression))
-                {
-                    diagnostics.Add(ErrorCode.ERR_IllegalSuppression, expression.Location);
-                    hasErrors = true;
-                }
-
                 var boundType = (BoundTypeExpression)convertedExpression;
                 bool isExplicitNotNullTest = boundType.Type.SpecialType == SpecialType.System_Object;
                 return new BoundTypePattern(node, boundType, isExplicitNotNullTest, inputType, boundType.Type, hasErrors);
             }
-
-            static bool hasSuppression(ExpressionSyntax e)
-            {
-                while (true)
-                {
-                    switch (e)
-                    {
-                        case ParenthesizedExpressionSyntax p:
-                            e = p.Expression;
-                            break;
-                        case PostfixUnaryExpressionSyntax { RawKind: (int)SyntaxKind.SuppressNullableWarningExpression }:
-                            return true;
-                        default:
-                            return false;
-                    }
-                }
-            }
         }
 
-        private ExpressionSyntax SkipParensAndNullSuppressions(ExpressionSyntax e)
+        private static ExpressionSyntax SkipParensAndNullSuppressions(ExpressionSyntax e, BindingDiagnosticBag diagnostics, ref bool hasErrors)
         {
             while (true)
             {
-                switch (e)
+                switch (e.Kind())
                 {
-                    case ParenthesizedExpressionSyntax p:
-                        e = p.Expression;
-                        break;
-                    case PostfixUnaryExpressionSyntax { RawKind: (int)SyntaxKind.SuppressNullableWarningExpression } p:
-                        e = p.Operand;
-                        break;
+                    case SyntaxKind.DefaultLiteralExpression:
+                        diagnostics.Add(ErrorCode.ERR_DefaultPattern, e.Location);
+                        hasErrors = true;
+                        goto default;
```

Comment:
I feel like the particular control flow here is a little weird. i.e. this could be `return e` and the `continue`s could be `break`s. However this seems subjective enough that I won't block the PR about it.